sdk/test-utils/recorder-new/src/utils/utils.ts

```diff
@@ -257,23 +256,62 @@ export interface RecorderStartOptions {
  *
  * Returns true if the param exists.
  */
-export function ensureExistence<T>(thing: T | undefined, label: string, mode: string): thing is T {
+export function ensureExistence<T>(thing: T | undefined, label: string): thing is T {
   if (!thing) {
     throw new RecorderError(
-      `Something went wrong, ${label} should not have been undefined in ${mode} mode.`
+      `Something went wrong, ${label} should not have been undefined in "${getTestMode()}" mode.`
     );
   }
   return true; // Since we would throw error if undefined
 }
 
+export type TestMode = "record" | "playback" | "live";
+
 /**
  * Returns the test mode.
  *
  * If TEST_MODE is not defined, defaults to playback.
  */
-export function getTestMode(): string {
+export function getTestMode(): TestMode {
   if (isPlaybackMode()) {
     return "playback";
   }
-  return env.TEST_MODE;
+  return env.TEST_MODE as "record" | "live";
+}
+
+/** Make a lazy value that can be deferred and only computed once. */
+export const once = <T>(make: () => T): (() => T) => {
+  let value: T;
+  return () => (value = value ?? make());
+};
+
+export function isRecordMode() {
+  return env.TEST_MODE === "record";
+}
+
+export function isLiveMode() {
+  return env.TEST_MODE === "live";
+}
+
+export function isPlaybackMode() {
+  return !isRecordMode() && !isLiveMode();
+}
+
+/**
+ * Loads the environment variables in both node and browser modes corresponding to the key-value pairs provided.
+ *
+ * Example-
+ *
+ * Suppose `variables` is { ACCOUNT_NAME: "my_account_name", ACCOUNT_KEY: "fake_secret" },
+ * `setEnvironmentVariables` loads the ACCOUNT_NAME and ACCOUNT_KEY in the environment accordingly.
+ * @export
+ * @param {{ [key: string]: string }} variables
+ */
+export function setEnvironmentVariables(
+  env: Record<string, string | undefined>,
+  variables: { [key: string]: string }
+) {
+  Object.keys(variables).map((key) => {
```

Comment:
nit: it feels weird using `map()` not for mapped output.  Maybe `forEach()`?

Reply:
updated 🙂